src/dvc_data/hashfile/tree.py

```diff
@@ -308,3 +312,23 @@ def merge(odb, ancestor_info, our_info, their_info):
     merged.digest()
 
     return merged
+
+
+def tree_from_index(
+    index: "DataIndex",
+    prefix: "DataIndexKey",
+) -> Tuple["Meta", "Tree"]:
```

Comment:
This could be `Tree.from_index` classmethod but I kept it separate since it returns the Meta, Tree tuple (vs just the Tree for typical `from_x` methods)